src/base/test/test_string.cc

```diff
@@ -122,4 +122,39 @@ BOOST_AUTO_TEST_CASE(trim_split_replace) {
 	BOOST_CHECK_EQUAL(str, "foo/wordr/wordz");
 }
 
+BOOST_AUTO_TEST_CASE(string_formatting) {
+	BOOST_CHECK_EQUAL("Hello World", bformat("%s", "Hello World"));
+	BOOST_CHECK_EQUAL("Hello World", bformat("%s %s", "Hello", "World"));
+	BOOST_CHECK_EQUAL("Hello World", bformat("%1$s %2%", "Hello", "World"));
+	BOOST_CHECK_EQUAL("Hello World", bformat("%2% %1%", "World", "Hello"));
+	BOOST_CHECK_EQUAL("   Hello World", bformat("%1$14s", "Hello World"));
+	BOOST_CHECK_EQUAL("Hello World   ", bformat("%-14s", "Hello World"));
+	BOOST_CHECK_EQUAL("         Hello", bformat("%14.5s", "Hello World"));
+	BOOST_CHECK_EQUAL("Hello         ", bformat("%1$-14.5s", "Hello World"));
+
+	BOOST_CHECK_EQUAL("A123X", bformat("A%dX", 123));
+	BOOST_CHECK_EQUAL("AABCDEFX", bformat("A%XX", 0xABCDEF));
+	BOOST_CHECK_EQUAL("A-1X", bformat("A%dX", -1));
+	BOOST_CHECK_EQUAL("A     123X", bformat("A%8dX", 123));
+	BOOST_CHECK_EQUAL("A00000123X", bformat("A%08dX", 123));
+	BOOST_CHECK_EQUAL("A0123X", bformat("A%d%u%d%uX", 0, 1, 2, 3));
```

Comment:
The `+` and `-` modifiers are not tested.
Neither for floats.

Reply:
Added tests for this, and also for various errors.